src/main/java/andrew/BuildingGame/Game/Game.java

```diff
@@ -220,4 +225,20 @@ private void gameCleanup() {
     }
 //    BGReady.unreadyAllParticipants();
   }
+
+  private Boolean isBuildingRound() {
+    return roundNumber % 2 == 0 && isNeitherInitNorInitPromptState() && !isFinalRound(roundNumber);
+  }
+
+  private Boolean isFinalBuildingRound() {
+    return isBuildingRound() && isFinalRound(roundNumber + 2);
+  }
+
+  private Boolean isFinalRound(int roundNumber) {
+    return roundNumber >= vars.getNumRounds();
+  }
+
+  private Boolean isNeitherInitNorInitPromptState() {
```

Comment:
Minor: I would rename to something easier to read + understand

e.g. isGameBeingInitialized()

Although naming it like this would mean you have to negate the usage above, but I think it is worth it - if you can think of a better name that can avoid negation for the usage, that would be even better.